swarm.go

```diff
@@ -103,11 +102,14 @@ type Swarm struct {
 // `extra` interface{} parameter facilitates the future migration. Supported
 // elements are:
 //  - connmgr.ConnectionGater
-func NewSwarm(ctx context.Context, local peer.ID, peers peerstore.Peerstore, bwc metrics.Reporter, extra ...interface{}) *Swarm {
+func NewSwarm(local peer.ID, peers peerstore.Peerstore, bwc metrics.Reporter, extra ...interface{}) *Swarm {
```

Comment:
If we're breaking this interface anyways, let's consider using functional options. We can also do that in a followup patch.